src/Facets/GenericSwapFacet.sol

```diff
@@ -3,42 +3,328 @@ pragma solidity 0.8.17;
 
 import { ILiFi } from "../Interfaces/ILiFi.sol";
 import { LibAsset } from "../Libraries/LibAsset.sol";
+import { LibSwap } from "../Libraries/LibSwap.sol";
 import { ReentrancyGuard } from "../Helpers/ReentrancyGuard.sol";
 import { SwapperV2, LibSwap } from "../Helpers/SwapperV2.sol";
 import { Validatable } from "../Helpers/Validatable.sol";
 import { LibUtil } from "../Libraries/LibUtil.sol";
-import { InvalidReceiver } from "../Errors/GenericErrors.sol";
+import { InvalidReceiver, ContractCallNotAllowed, CumulativeSlippageTooHigh, NativeAssetTransferFailed } from "../Errors/GenericErrors.sol";
+import { IERC20, SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { LibAllowList } from "../Libraries/LibAllowList.sol";
+import { console2 } from "forge-std/console2.sol";
 
-/// @title Generic Swap Facet
+/// @title GenericSwapFacet
 /// @author LI.FI (https://li.fi)
 /// @notice Provides functionality for swapping through ANY APPROVED DEX
-/// @dev Uses calldata to execute APPROVED arbitrary methods on DEXs
-/// @custom:version 1.0.0
+/// @dev Can only execute calldata for APPROVED function selectors
+/// @custom:version 2.0.0
 contract GenericSwapFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable {
+    using SafeERC20 for IERC20;
```

Comment:
Why don't we use `import { ERC20, SafeTransferLib } from "solmate/utils/SafeTransferLib.sol";` as in the packed facets?

Reply:
We use both implementations in various places. Never thought about it really. 
I ran some tests...the solmate implementation is more gas-efficient. I will change it and also write down that we will only use solmate for future implementations. 
Good feedback, thank you !